common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -740,6 +746,61 @@ def associate_by_email_if_login_api(auth_entry, backend, details, user, current_
             return association_response
 
 
+@partial.partial
+def associate_by_email_if_saml(auth_entry, backend, details, user, strategy, *args, **kwargs):
+    """
+    This pipeline step associates the current social auth with the user with the
+    same email address in the database.  It defers to the social library's associate_by_email
+    implementation, which verifies that only a single database user is associated with the email.
+
+    This association is done ONLY if the user entered the pipeline belongs to SAML provider.
+    """
+    # pylint: disable=too-many-nested-blocks
+    if is_multiple_sso_accounts_association_to_saml_user_enabled():
+
+        provider_saml, current_provider = is_provider_saml(strategy.request.backend.name, kwargs)
+
+        if provider_saml:
+
+            # get the user by matching email if the pipeline user is not available.
+            current_user = user
+
+            if not current_user:
+                user_details = {'email': details.get('email')} if details else None
+                current_user = get_user(user_details or {})
+
+            if current_user:
```

Comment:
Is there a possible scenario where the `if current user` check returns `false`, and if so, how should that be handled?

Reply:
Yes, that's possible in case when the user is coming for the very first time and doesn't have an account on edX. In that case, everything is working fine already.